src/main.cpp

```diff
@@ -166,10 +166,25 @@ namespace
 			//     {"type": "user",  "name": "jane#otherZone"}
 			//   ]
 			// }
+
+			// Per the API, any of arg5 to arg8 is eligible to be the new attribute
+			// (depending on which one starts with "n:")
+			// However, only one will be considered (multiples will throw CAT_INVALID_ARGUMENT)
+			// Therefore, find the first one that starts with "n:"
+			const auto candidates = {input->arg5, input->arg6, input->arg7, input->arg8};
```

Comment:
Can you explain why it's four inputs instead of three?

Is it because the units input for the source metadata is optional?

Reply:
Referencing https://github.com/irods/irods/blob/e6df49f58759d6609650b82ce4a907d82ee10881/plugins/database/src/db_plugin.cpp#L9444 here, the arguments get passed in top-to-bottom. arg5 is `_unitsOrArg0` which is eligible to be one of the `n:` 
https://github.com/irods/irods/blob/e6df49f58759d6609650b82ce4a907d82ee10881/plugins/database/src/db_plugin.cpp#L9485
...if its type is considered as such. 
https://github.com/irods/irods/blob/e6df49f58759d6609650b82ce4a907d82ee10881/plugins/database/src/db_plugin.cpp#L9499
This shows that those 4 can potentially be the `n:`, so we check all 4 of them. 
https://github.com/irods/irods/blob/e6df49f58759d6609650b82ce4a907d82ee10881/plugins/database/src/db_plugin.cpp#L9502-L9506
This shows that CAT_INVALID_ARGUMENT is thrown if you try to pass `n:` more than once, so just finding the first one should be good enough.